aggregator/src/binaries/janus_cli.rs

```diff
@@ -155,6 +341,95 @@ async fn install_tracing_and_metrics_handlers(
     Ok((trace_guard, metrics_guard))
 }
 
+async fn generate_global_hpke_key<C: Clock>(
+    datastore: &Datastore<C>,
+    dry_run: bool,
+    id: HpkeConfigId,
+    kem: HpkeKemId,
+    kdf: HpkeKdfId,
+    aead: HpkeAeadId,
+    hpke_config_out_file: Option<&Path>,
+) -> Result<()> {
+    let hpke_keypair = Arc::new(generate_hpke_config_and_private_key(id, kem, kdf, aead)?);
+
+    if !dry_run {
+        datastore
+            .run_tx("generate-global-hpke-key", |tx| {
+                let hpke_keypair = Arc::clone(&hpke_keypair);
+
+                Box::pin(async move { tx.put_global_hpke_keypair(&hpke_keypair).await })
+            })
+            .await?;
+    }
+
+    if let Some(hpke_config_out_file) = hpke_config_out_file {
+        fs::write(hpke_config_out_file, hpke_keypair.config().get_encoded()?).await?;
+    }
+
+    Ok(())
+}
+
+async fn set_global_hpke_key_state<C: Clock>(
+    datastore: &Datastore<C>,
+    dry_run: bool,
+    id: HpkeConfigId,
+    state: HpkeKeyState,
+) -> Result<()> {
+    if !dry_run {
+        datastore
+            .run_tx("set-global-hpke-key-state", |tx| {
+                Box::pin(async move { tx.set_global_hpke_keypair_state(&id, &state).await })
+            })
+            .await?;
+    }
+
+    Ok(())
+}
+
+async fn add_taskprov_peer_aggregator<C: Clock>(
+    datastore: &Datastore<C>,
+    dry_run: bool,
+    peer_endpoint: &Url,
+    role: Role,
+    verify_key_init: VerifyKeyInit,
+    collector_hpke_config_file: &Path,
+    report_expiry_age: Option<Duration>,
+    tolerable_clock_skew: Duration,
+    aggregator_auth_token: &AuthenticationToken,
+    collector_auth_token: Option<&AuthenticationToken>,
+) -> Result<()> {
+    let collector_hpke_config = {
+        let bytes = fs::read(collector_hpke_config_file).await?;
+        HpkeConfig::get_decoded(&bytes)?
+    };
+    let collector_auth_tokens = collector_auth_token
+        .cloned()
+        .map(|token| Vec::from([token]))
+        .unwrap_or_default();
+    let peer_aggregator = Arc::new(PeerAggregator::new(
+        peer_endpoint.clone(),
+        role,
+        verify_key_init,
+        collector_hpke_config,
+        report_expiry_age,
+        tolerable_clock_skew,
+        Vec::from([aggregator_auth_token.clone()]),
+        collector_auth_tokens,
+    ));
+
+    if !dry_run {
+        datastore
+            .run_tx("add-taskprov-peer-aggregator", |tx| {
+                let peer_aggregator = Arc::clone(&peer_aggregator);
+
+                Box::pin(async move { tx.put_taskprov_peer_aggregator(&peer_aggregator).await })
+            })
+            .await?;
+    }
+
+    Ok(())
+}
+
```

Comment:
We may also want to plumb through `delete_taskprov_peer_aggregator`, since if we make a mistake in the configuration we'll otherwise have to go into the DB to drop it.

Reply:
Agreed, but I'm going to leave this for a later PR. If we make a mistake during configuration, we really can hop into the DB without much risk as the relevant deployment will not be used until after configuration is complete.